src/test/kotlin/reminder/application/ReminderDeleterTest.kt

```diff
@@ -0,0 +1,52 @@
+package reminder.application
+
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertDoesNotThrow
+import org.junit.jupiter.api.assertThrows
+import org.mockito.Mockito
+import reminder.domain.ReminderIdentifierGenerator
+import reminder.domain.ReminderRepository
+import reminder.domain.exceptions.NonExistentReminderException
+import reminder.mothers.IdentifierMother
+import reminder.mothers.ReminderMother
+import java.util.*
+
+class ReminderDeleterTest {
+    private lateinit var reminderDeleter: ReminderDeleter
+    private lateinit var generator: ReminderIdentifierGenerator
+    private lateinit var repository: ReminderRepository
+
+    @BeforeEach
+    fun setUp() {
+        repository = Mockito.mock(ReminderRepository::class.java)
+        generator = Mockito.mock(ReminderIdentifierGenerator::class.java)
+        reminderDeleter = ReminderDeleter(repository)
+    }
+
+    @Test
+    fun `Invalid identifier throws NonExistentReminderException`() {
+        // Initialize
+        val identifier = IdentifierMother.getValidIdentifier()
+        Mockito.`when`(repository.search(identifier)).thenReturn(Optional.empty())
+        // Execute and Assert
+        assertThrows<NonExistentReminderException> {
+            reminderDeleter.delete(identifier)
+        }
+        // Assert number of calls
+        Mockito.verify(repository, Mockito.times(0)).delete(identifier)
+    }
+
+    @Test
+    fun `Nothing is returned when called with existent identifier`() {
```

Comment:
Maybe: "if the reminder exists, it delete its from the repository" would be a better naming? (Yes I am super picky with this stuff 😢 )

Reply:
![](https://media.tenor.com/-2nRJSPeMg0AAAAC/snob-mimimi.gif)